code_samples/back_office/menu/menu_item/src/EventSubscriber/MyMenuSubscriber.php

```diff
@@ -5,22 +5,39 @@
 use Ibexa\AdminUi\Menu\Event\ConfigureMenuEvent;
 use Ibexa\AdminUi\Menu\MainMenuBuilder;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Ibexa\AdminUi\Menu\MenuItemFactory;
 
 class MyMenuSubscriber implements EventSubscriberInterface
 {
+    private $menuItemFactory;
+
+    public function __construct(MenuItemFactory $menuItemFactory)
+    {
+        $this->menuItemFactory = $menuItemFactory;
+    }
+
     public static function getSubscribedEvents()
     {
         return [
             ConfigureMenuEvent::MAIN_MENU => ['onMainMenuConfigure', 0],
-            ConfigureMenuEvent::CONTENT_SIDEBAR_RIGHT => ['onContentSidebarConfigure', 0]
         ];
     }
 
     public function onMainMenuConfigure(ConfigureMenuEvent $event)
     {
         $menu = $event->getMenu();
 
-        $menu[MainMenuBuilder::ITEM_CONTENT]->addChild(
+        $custom_menu_item = $menu[MainMenuBuilder::ITEM_CONTENT]->addChild(
+            'main__content__custom_menu',
+            [
+                'extras' => [
+                    'orderNumber' => 100,
+                ],
+            ],
+        );
+
+        $custom_menu_item->addChild(
+            $this->menuItemFactory->createItem(
```

Comment:
Since factory assignment has been properly added to each menu `Item`, there is no longer need to call `$this->menuItemFactory->createItem()` here. Items can be added just like in the example above - by passing the same arguments as those that are usually passed into factory.

```php
$custom_menu_item->addChild(
    'all_content_list',
    [
        'label' => 'Content List',
        'route' => 'all_content_list.list',
        'attributes' => [
            'class' => 'custom-menu-item',
        ],
        'linkAttributes' => [
            'class' => 'custom-menu-item-link',
        ],
    ]
);
```

Additionally, variable names are commonly created using `camelCase` by convention. Using `snake_case` is usually reserved for array keys and twig.